tests/admin_views/test_multidb.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import patch
+
+from django.conf.urls import url
+from django.contrib import admin
+from django.contrib.auth.models import User
+from django.db import connections, router
+from django.test import TestCase, override_settings
+
+from .models import Book
+
+
+class Router(object):
+    def __init__(self, target):
+        self.target = target
+
+    def db_for_read(self, model, **hints):
+        return self.target
+
+    def db_for_write(self, model, **hints):
+        return self.target
+
+site = admin.AdminSite(name="test_adminsite")
+site.register(Book)
+
+urlpatterns = [
+    url(r'^admin/', site.urls),
+]
+
+
+@override_settings(ROOT_URLCONF=__name__)
+class MultiDatabaseTest(TestCase):
+    multi_db = True
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.superuser = {}
+        for db in connections:
+            router.routers = [Router(db)]
```

Comment:
You'll want to store the original routers and restore them in `tearDownClass` to preserve test isolation.